site/content/entry/citra-progress-report-2023-q1/index.md

```diff
@@ -0,0 +1,413 @@
+
+By removing all fully invalidated surfaces and implementing a map of recycled textures from destroyed surfaces to avoid driver overhead from allocations, performance on poorly behaved games has been greatly increased.
+
+{{< figure src="pilotwings.png"
+    title="A great boost in performance for problematic games!" >}}
+
+### Add svcGetSystemInfo 0x20000 to get Citra information ([#5955](https://github.com/citra-emu/citra/pull/5955)) by [PabloMK7](https://github.com/PabloMK7)
+
+`svcGetSystemInfo` is used by 3DS applications to gather the information of a system, such as whether someone is playing on a 3DS or 2DS, and to communicate this information back to the application. Homebrew apps also use this for the same purposes. 
+As homebrew apps had no way of telling whether they were running on hardware or on Citra, this implements a way for those aforementioned homebrew apps to finally tell if they are being run on Citra or not!
+
+### Allow MemoryRef to hold a past-the-end offset ([#6141](https://github.com/citra-emu/citra/pull/6141)) by [vitor-k](https://github.com/vitor-k)
+
+Previously, Citra did not correctly understand any copies or operations being held at the end of the emulated VRAM, which would cause them to get completely ignored. This would cause numerous errors in the emulation of games, including some graphics being cut out entirely on HarmoKnight. By allowing MemoryRef to correctly understand these values, they will no longer be ignored and can be executed correctly.
+
+{{< sidebyside "image" ""
+    "harmoOld.png=What's happened there?"
+    "harmoNew.png=Ah, that's better!" >}}
+
+### arm/dynarmic: More accurate cycle counting ([#6194](https://github.com/citra-emu/citra/pull/6194)) by [merryhime](https://github.com/merryhime)
+
+While most games easily run at full speed nowadays, a long running issue in Citra has been pre-rendered cutscene videos running way too slowly. This has always been a bit of a mystery, since decoding a video at a resolution of 240p shouldn’t be too taxing on the system of an average user.
+
+By implementing more accurate cycle counting, we are happy to announce we have made a first step towards solving this problem, improving FMV video performance by up to 3x! To understand what this means, we need to take a closer look at how a processor works.
+
+On the lowest level, all that a CPU does is process a list of instructions. Some instructions however take longer to execute than others. The time it takes for an instruction to execute is measured in cycles. Previously, we were using a very naive implementation of assuming every instruction would take one cycle. This leads to games idling longer than they should, therefore unnecessarily slowing down the emulation.
+Now [merryhime](https://github.com/merryhime), the author of [dynarmic](https://github.com/merryhime/dynarmic), has implemented much more accurate cycle counts, significantly reducing this issue.
```

Comment:
You say that this is a first step, but that performance is improved by 3X. If videos are still problematic, there should be another sentence or so explaining how so.

Reply:
So uhhh, I recall that we don't really know why it runs as badly as it does. So that would be hard to explain.

Reply:
Fair, I am just pointing out that at plain surface it's a little confusing. As a naïve user, I would assume that a 3X improvement meant the problem is largely solved. If it is not, a one-liner saying that "performance is still problematic due to unknown reasons, expect videos to run half speed on our minimum recommended hardware" or something would set expectations a little better.